src/plan.rs

```diff
@@ -294,13 +303,88 @@ impl Plan {
     }
 }
 
+#[derive(Debug)]
+/// Signatures which a key can produce
+///
+/// Defaults to `ecdsa=true` and `taproot=TaprootCanSign::default()`
+pub struct CanSign {
+    /// Whether the key can produce ECDSA signatures
+    ecdsa: bool,
+    /// Whether the key can produce taproot (Schnorr) signatures
+    taproot: TaprootCanSign,
+}
+
+impl Default for CanSign {
+    fn default() -> Self {
+        CanSign {
+            ecdsa: true,
+            taproot: TaprootCanSign::default(),
+        }
+    }
+}
+
+#[derive(Debug)]
+/// Signatures which a taproot key can produce
+///
+/// Defaults to `key_spend=true`, `script_spend=Any` and `sighash_default=true`
+pub struct TaprootCanSign {
```

Comment:
Same comment about public fields.

Reply:
It is not clear to me if this is resolved, or not. @sanket1729 can you confirm you are happy with the visibility of these types please.